### PR TITLE
feat: add support for TLS auth on Redact storer

### PR DESCRIPTION
**Pull request by ajpauwels**
- Adds a config that allows a consuming binary to externally provide to any Redact storer a default client TLS config
- Adds the provided client TLS config to the http client configuration
- Properly loads CA certificates and verifies TLS connection
- Only supports single-cert use currently

### CODE REVIEW

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **ajpauwels**
You can assign the PR to them by writing `/assign @ajpauwels` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-crypto).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/pauwels-labs/redact-crypto/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/storage/redact.rs
Comment (howhelan) [inline src/storage/redact.rs:162]: what's the expected behavior if there is no config? Will this still work or should it just throw an error?

File: src/storage/redact.rs
Comment (howhelan) [inline src/storage/redact.rs:131]: worth adding some unit tests here?

File: src/storage/redact.rs
Comment (ajpauwels) [reply src/storage/redact.rs:162]: It will work but requests will be anonymous, i.e. will not have client TLS. So requests would fail if they need to be authenticated.

**Review by howhelan — APPROVED** [review]

**Comment by capt-haddock** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[howhelan](https://github.com/pauwels-labs/redact-crypto/pull/10#pullrequestreview-765381097 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=pauwels-labs%2Fredact-crypto).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/pauwels-labs/redact-crypto/blob/main/OWNERS)~~ [howhelan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
